RNWCPP/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +30,289 @@ enum class ScrollViewCommands
+
+  switch (commandId)
+  {
+    case static_cast<int64_t>(ScrollViewCommands::ScrollTo) :
+    {
+      double x = commandArgs[0].asDouble();
+      double y = commandArgs[1].asDouble();
+      bool animated = commandArgs[2].asBool();
+      bool horiz = scrollViewer.HorizontalScrollMode() == winrt::ScrollMode::Auto;
+      if (horiz)
+      {
+        // snap the x coordinate
+        if (m_isSnapToAlignmentSet && m_snapToInterval > 0.0)
+        {
+          x = snap(x, m_snapToAlignment, m_snapToInterval, scrollViewer.ViewportWidth());
+        }
+        // ChangeView parameters: x, y, zoom, animated
+        scrollViewer.ChangeView(x, nullptr, nullptr, !animated/*disableAnimation*/);
+      }
+      else
+      {
+        // snap the y coordinate
+        if (m_isSnapToAlignmentSet && m_snapToInterval > 0.0)
+        {
+          y = snap(y, m_snapToAlignment, m_snapToInterval, scrollViewer.ViewportHeight());
+        }
+        // ChangeView parameters: x, y, zoom, animated
+        scrollViewer.ChangeView(nullptr, y, nullptr, !animated/*disableAnimation*/);
+      }
+      break;
+    }
+
+    case static_cast<int64_t>(ScrollViewCommands::ScrollToEnd) :
+    {
+      bool animated = commandArgs[0].asBool();
+      bool horiz = scrollViewer.HorizontalScrollMode() == winrt::ScrollMode::Auto;
+      if (horiz)
+        scrollViewer.ChangeView(scrollViewer.ScrollableWidth(), nullptr, nullptr, !animated/*disableAnimation*/);
+      else
+        scrollViewer.ChangeView(nullptr, scrollViewer.ScrollableHeight(), nullptr, !animated/*disableAnimation*/);
+      break;
+    }
+  }
+}
+
+void ScrollViewShadowNode::createView()
+{
+  Super::createView();
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  ScrollViewManager* viewManager = static_cast<ScrollViewManager*>(GetViewManager());
+  int64_t tag = m_tag;
+
+  scrollViewer.ViewChanging([this, tag](const winrt::IInspectable& sender, const winrt::ScrollViewerViewChangingEventArgs& args)
+  {
+    winrt::ScrollViewer scrollViewer = sender.as<winrt::ScrollViewer>();
+    topScroll(scrollViewer, tag,
+      "topScroll",
+      args.NextView().HorizontalOffset(),
+      args.NextView().VerticalOffset(),
+      args.NextView().ZoomFactor());
+
+  });
+}
+ 
+
+void ScrollViewShadowNode::updateProperties(const folly::dynamic&& props)
+{
+  m_updating = true;
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  if (scrollViewer == nullptr)
+    return;
+
+  for (auto& pair : props.items())
+  {
+    const folly::dynamic& propertyName = pair.first;
+    const folly::dynamic& propertyValue = pair.second;
+
+    if (propertyName.asString() == "horizontal")
+    {
+      if (propertyValue.isBool())
+      {
+        bool horiz = propertyValue.getBool();
+
+        scrollViewer.HorizontalScrollMode(horiz ? winrt::ScrollMode::Auto : winrt::ScrollMode::Disabled);
+        scrollViewer.VerticalScrollMode(horiz ? winrt::ScrollMode::Disabled : winrt::ScrollMode::Auto);
+      }
+    }
```

Comment:
The docs say this controls which direction the ScrollView's children are arranged.  It doesn't say which directions are supposed to be scrollable, which leads to some questions:
-If you zoom in and the content no longer fits in either direction, are you supposed to be able to scroll in both directions to see the clipped content?
-If a child is allowed to exceed the boundaries of the ScrollView, same question?
If you're always supposed to be able to scroll to reveal any content that doesn't fit, I think you would want to set both HorizontalScrollMode and VerticalScrollMode to Auto unconditionally.